website/src/common/MacroBenchmarkTable.tsx

```diff
@@ -0,0 +1,207 @@
+  vtgateCpuTime: MacroBenchmarkTableDataRow;
+  vttabletCpuTime: MacroBenchmarkTableDataRow;
+  totalComponentsMemStatsAllocBytes: MacroBenchmarkTableDataRow;
+  vtgateMemStatsAllocBytes: MacroBenchmarkTableDataRow;
+  vttabletMemStatsAllocBytes: MacroBenchmarkTableDataRow;
+};
+
+export type MacroBenchmarkTableProps = {
+  data: MacroBenchmarkTableData;
+  oldGitRef: string;
+  newGitRef: string;
+};
+
+const getDeltaBadgeVariant = (key: string, delta: number, p: number) => {
+  if (delta === 0) {
+    return "warning";
+  }
+  if (
+    key.includes("CpuTime") ||
+    key.includes("Mem") ||
+    key.includes("latency")
+  ) {
+    if (delta < 0) {
+      return "success";
+    }
+  }
+  if (key.includes("qps") || key.includes("tps")) {
+    if (delta > 0) {
+      return "success";
+    }
+  }
+  return "destructive";
+};
+
+const formatCellValue = (key: string, value: any) => {
+  if (key.includes("CpuTime")) {
+    return secondToMicrosecond(value);
+  } else if (key.includes("MemStatsAllocBytes")) {
+    return formatByte(value);
+  }
+  return value;
+};
+
+export function getRange(range: Range) {
+  if (range.infinite == true) {
+    return "∞";
+  }
+  if (range.unknown == true) {
+    return "?";
+  }
+  return "±" + fixed(range.value, 1) + "%";
+}
+
+// Type-safe function to access properties
+const getValue = <T, K extends keyof T>(obj: T, key: K): T[K] => obj[key];
+
+export default function MacroBenchmarkTable({
+  data,
+  newGitRef,
+  oldGitRef,
+}: MacroBenchmarkTableProps) {
+  if (!data) {
+    return null;
+  }
+  const dataKeys = Object.keys(data) as Array<keyof MacroBenchmarkTableData>;
+
+  // Use keyof MacroBenchmarkTableData to enforce key types
+  const classNameMap: { [key in keyof MacroBenchmarkTableData]: string } = {
+    qpsTotal: "border-b border-foreground dark:border-none",
+    qpsReads: "bg-muted/80",
+    qpsWrites: "bg-muted/80",
+    qpsOther: "bg-muted/80 border-b border-foreground dark:border-none",
+    tps: "bg-background",
+    latency: "bg-background",
+    errors: "bg-background",
+    totalComponentsCpuTime: "border-b border-foreground dark:border-none",
+    vtgateCpuTime: "bg-muted/80 ",
+    vttabletCpuTime:
+      "bg-muted/80 border-b border-foreground dark:border-none",
+    totalComponentsMemStatsAllocBytes:
+      "border-b border-foreground dark:border-none",
+    vtgateMemStatsAllocBytes: "bg-muted/80",
+    vttabletMemStatsAllocBytes:
+      "bg-muted/80 border-b border-foreground dark:border-none",
+  };
+
+  return (
+    <Table>
+      <TableHeader>
+        <TableRow className="hover:bg-background border-b">
+          <TableHead className="w-[200px]"></TableHead>
+          <TableHead className="text-center text-primary font-semibold">
+            {formatGitRef(oldGitRef) || "N/A"}
+          </TableHead>
+          <TableHead className="text-center text-primary font-semibold">
+            {formatGitRef(newGitRef) || "N/A"}
+          </TableHead>
```

Comment:
Let's make the git ref clickable so they redirect to the commit on the vitess repo